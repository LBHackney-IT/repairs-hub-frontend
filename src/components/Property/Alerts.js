import PropTypes from 'prop-types'
import TruncateText from '../Layout/TruncateText'

const Alerts = ({ alerts, setIsExpanded, isExpanded }) => {
  return (
    <>
      {alerts.map((alert, index) => {
        return (
          <li
            className="bg-orange"
            style={{
              padding: '0.5rem',
              overflow: 'hidden',
              borderRadius: '10px',
            }}
            key={index}
          >
            <p style={{ marginTop: 0 }}>
              {alert.comments} <strong>{alert.type}</strong>
            </p>
            {alert.comments === 'Specific Requirements' && (
              <TruncateText
                text={alert.reason}
                numberOfLines="3"
                pTagClassName="lbh-body-s text-!-white"
                linkClassName="lbh-body-xs text-!-white"
                setIsExpanded={setIsExpanded}
                isExpanded={isExpanded}
              />
            )}
          </li>
        )
      })}
    </>
  )
}

Alerts.propTypes = {
  alerts: PropTypes.arrayOf(
    PropTypes.shape({
      comments: PropTypes.string,
      reason: PropTypes.string,
      startDate: PropTypes.string,
      endDate: PropTypes.string,
      alertId: PropTypes.string,
    })
  ).isRequired,
  setIsExpanded: PropTypes.fun,
  isExpanded: PropTypes.bool,
}

export default Alerts
