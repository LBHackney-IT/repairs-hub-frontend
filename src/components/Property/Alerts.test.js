import { render } from '@testing-library/react'
import Alerts from './Alerts'

describe('Alerts component', () => {
  it('should render only comment and type when there are not any Specific Requirements alerts', () => {
    const alerts = [
      {
        type: 'OIT',
        comments: 'Other',
        startDate: '12/12/2024',
        endDate: null,
        reason: 'Test',
        alertId: 'a821cd48-8664-4d43-81ab-d5baf689e0c7',
      },
      {
        type: 'OIT',
        comments: 'Hello',
        startDate: '12/12/2024',
        endDate: null,
        reason: 'New Test',
        alertId: 'a821cd48-8664-4d43-81ab-d5baf689e0c9',
      },
    ]
    const { asFragment } = render(
      <Alerts alerts={alerts} setIsExpanded={jest.fn()} isExpanded={false} />
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it('should render comment, type and reason for a Specific Requirement alert and comment and type for other alerts', () => {
    const alerts = [
      {
        type: 'OIT',
        comments: 'Other',
        startDate: '12/12/1212',
        endDate: null,
        reason: 'Test',
        alertId: 'a821cd48-8664-4d43-81ab-d5baf689e0c7',
      },
      {
        type: 'SPR',
        comments: 'Specific Requirements',
        startDate: '09/01/2025',
        endDate: null,
        reason:
          'Taking up the whole of the characters to see what it looks like Taking up the whole of the characters to see what it looks likeTaking up the whole of the characters to see what it looks likeTaking up the whole of the characters to see what it looks likeTaking up the whole of the characters to see what it looks likeTaking up the whole of the characters to see what it looks likeTaking up the whole of the characters to see what it looks likeTaking up the whole of the characters to see what it looks likeTaking up the whole of the characters to see what it looks likeTaking up the whole of the characters to see what it looks likeTaking up the whole of the characters to see what it looks likeTaking up the whole of the characters to see what it looks likeTaking up the whole of the characters to see what it looks likeTaking up the whole of the characters to see what it looks likeTaking up the whole of the characters to see what it looks likeTaking up the whole of the characters to see what it l',
        alertId: '73cf5a20-628e-47a0-9c42-c814e307baf5',
      },
    ]

    const { asFragment } = render(
      <Alerts alerts={alerts} setIsExpanded={jest.fn()} isExpanded={false} />
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
