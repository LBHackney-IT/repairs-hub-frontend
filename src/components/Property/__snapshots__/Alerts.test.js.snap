// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alerts component should render comment, type and reason for a Specific Requirement alert and comment and type for other alerts 1`] = `
<DocumentFragment>
  <li
    class="bg-orange"
    style="padding: 0.5rem; overflow: hidden; border-radius: 10px;"
  >
    <p
      style="margin-top: 0px;"
    >
      Other 
      <strong>
        (OIT)
      </strong>
    </p>
  </li>
  <li
    class="bg-orange"
    style="padding: 0.5rem; overflow: hidden; border-radius: 10px;"
  >
    <p
      style="margin-top: 0px;"
    >
      Specific Requirements 
      <strong>
        (SPR)
      </strong>
    </p>
    <p
      class="lbh-body-s text-!-white truncate-description truncate-line-3"
    >
      Specific requirement alert
    </p>
  </li>
</DocumentFragment>
`;

exports[`Alerts component should render only comment and type when there are not any Specific Requirements alerts 1`] = `
<DocumentFragment>
  <li
    class="bg-orange"
    style="padding: 0.5rem; overflow: hidden; border-radius: 10px;"
  >
    <p
      style="margin-top: 0px;"
    >
      Other 
      <strong>
        (OIT)
      </strong>
    </p>
  </li>
  <li
    class="bg-orange"
    style="padding: 0.5rem; overflow: hidden; border-radius: 10px;"
  >
    <p
      style="margin-top: 0px;"
    >
      Hello 
      <strong>
        (OIT)
      </strong>
    </p>
  </li>
</DocumentFragment>
`;
